### ISSUE TITLE
Cover grid hides entire albums after filtering one track

### ISSUE THREAD

**Issue by IBBoard** [quodlibet/quodlibet#2345]
Coming from other media players, I like having the cover grid, but I'm used to it being a collection of all of the cover art for the albums of the tracks that match my current search. I can't seem to make Quod Libet behave like that.

Steps to recreate:
 1. Create a playlist called "SpecialOccasions"
 1. Add "Track 1" from "Album A" to the playlist
 1. Go to Cover Grid
 1. Search/filter with "!~playlists = "SpecialOccasions" to show every-day tracks

Expected behaviour:
 1. "Album A" appears in cover list
 1. "Album A" contains all tracks *except* "Track 1"

Actual behaviour:
 1. "Album A" does not appear in cover list at all

The same happens when filtering by artist. If I don't want to listen to one artist today and filter them out, but they've got a track on a compilation, then the *whole* compilation is removed. That could be useful in some situations, but doesn't feel like the most useful default behaviour.

(I'm not sure whether this is a bug report, a feature request or a documentation request!)

Versions: Quod Libet 3.8.1 on openSUSE 42.2 using the "official" repo

**Comment by declension** [comment]
If I've understood you correctly, this is expected behaviour for album browsers (not limited to the Cover Grid):
 * Albums are collections which define songlike aggregate views onto their songs
 * These are aggregated (reduced) by a variety of operators depending on tag, with sensible defaults (e.g. average for `~bitrate`, sum for `~length` min for dates, or set union for most other tags (including say `~playlists`))

Essentially: 
 * Your example then is really asking _Show me **albums** which don't have any song belonging to the playlist 'SpecialOccasions'_.
 * What you and other players are using is _Show me **songs** which don't belong to the playlist 'SpecialOccasions' and then group them by album (removing if there are none)_

I don't see this working in those QL browsers without rethinking how all of them operate. The most realistic way I guess would be to allow text aggregation functions and define a new one (as well as a default e.g. `union`) which was more like an intersection function on sets. Then it could be used like this:

`~playlists:inter = SpecialOccasions'`

**Comment by IBBoard** [comment]
Yes, that's a better description. From what you've said then it sounds like there's nothing I can do with a search to get the behaviour I want, which is a shame, and that this is a feature request.

**Comment by declension** [comment]
I understand, this is confusing at best (technically it's the fact that one track _does_ match the query, meaning the album does, before it's then negated...)

Currently, I guess it's best to use a songs browser (i.e. _Search_ / _Paned_) for this use-case. There are discussions around about aggregation that I started in the slightly dead-end #821 and probably a few elsewhere too; the _Duplicates Browser_ plugin has a rudimentary grouping feature which in theory could display album art.

For this issue then I'll convert it to a feature request as you suggest, as that might be a faster route.
